warehouse/query-core/src/main/java/datawave/query/jexl/visitors/TreeFlatteningRebuildingVisitor.java

```diff
@@ -70,44 +72,78 @@ public Object visit(ASTOrNode node, Object data) {
         ASTOrNode orNode = JexlNodes.newInstanceOfType(node);
         orNode.image = node.image;
         
-        return flattenTree(node, orNode, data);
+        LinkedList<JexlNode> stackA = new LinkedList<>();
```

Comment:
Just a thought, but should we do the same for AND nodes?  I realize that very large OR nodes may be more common so I could go without that.